migrator/migrator.go

```diff
@@ -92,14 +105,19 @@ func (m Migrator) FullDataTypeOf(field *schema.Field) (expr clause.Expr) {
 // AutoMigrate auto migrate values
 func (m Migrator) AutoMigrate(values ...interface{}) error {
 	for _, value := range m.ReorderModels(values, true) {
-		tx := m.DB.Session(&gorm.Session{})
-		if !tx.Migrator().HasTable(value) {
-			if err := tx.Migrator().CreateTable(value); err != nil {
+		queryTx := m.DB.Session(&gorm.Session{})
+		execTx := queryTx
+		if m.DB.DryRun {
+			queryTx.DryRun = false
+			execTx = m.DB.Session(&gorm.Session{Logger: &printSQLLogger{}})
```

Comment:
the `printSQLLogger` should reuse the current logger to print the running SQL?

Reply:
The existing logger print some tracing text, so we can not copy the output to execute it.
I think a new logger `printSQLLogger` is necessary.

Reply:
Any other suggestion on this PR? @jinzhu